pkgs/applications/blockchains/monero-cli/default.nix

```diff
@@ -112,14 +109,24 @@ stdenv.mkDerivation rec {
       "-DCMAKE_CXX_FLAGS=-fpermissive"
     ];
 
-  outputs = [ "out" "source" ];
+  outputs = [
+    "out"
+    "source"
+  ];
 
   meta = {
     description = "Private, secure, untraceable currency";
     homepage = "https://getmonero.org/";
     license = lib.licenses.bsd3;
-    platforms = lib.platforms.all;
-    maintainers = with lib.maintainers; [ rnhmjoj ];
+
+    # linux is known to work fully.
+    # macOS/ARM has a working `monerod` (at least), but `monero-wallet-cli` segfaults on start after entering the wallet password.
+    platforms = with lib.platforms; darwin ++ linux;
```

Comment:
If the wallet is unusable it's probably better to just mark monero as broken: I don't see the point of distributing a half broken program.

Is this only for aarch64-darwin or also x86_64-darwin? I believe at some point it was tested and used to work on the latter.

Reply:
I have only aarch64-darwin, so that's all I tested with.

Yes, I think you're right that it's better to turn off darwin support entirely until the wallet CLI works. I will update it.

Reply:
So, le'ts recap: before updating the submodule aarch64-darwin wouldn't build, after it does build, but the wallet segfaults pretty soon. Is this even with trezor turned off?

Reply:
Yes, the package in master now does not build on my M1.

Now it builds and the `monerod` daemon appears to work: it reads my existing datadir and syncs. But the wallet binary segfaults on startup.

Tangentially, the *Homebrew* version (also 0.18.3.4) also segfaults, but on exit, so at least I am able to operate the wallet. But there is something going on that's probably not specific to Nixpkgs.

Reply:
Uhm, ok then I would keep `trezorSupport ? true` and set `meta.badPlatform = [ "aarch64-darwin" ]`, for now.
If you can do a debug build with symbols, try making a report upstream.

Reply:
Today I started running into the same error I ran into yesterday - and I can't remember how I got around it yesterday.

Now monero doesn't build on my mac at all, whether I am building through Nixpkgs or directly with Homebrew dependencies.

I opened a report upstream: https://github.com/monero-project/monero/issues/9486

I will close this PR for now til I (re)figure out how to build this package on my Mac.

Reply:
I figured out how I got around it - the error happens only when building in *debug* mode. I was trying to build in debug mode so I can report the segfault problem.